lib/convection/control/stack.rb

```diff
@@ -170,6 +162,42 @@ def cloud_name
         "#{ cloud }-#{ name }"
       end
 
+      # Calls the Convection::Model::Stack#resolver method
+      # if there is a rate limiting exception it sleeps and trys again.
+      def resolver_caller(count = 0.25)
+        resolver
+      rescue Aws::CloudFormation::Errors::Throttling
+        sleep count
+        count += 0.25
+        resolver_caller(count)
+      end
+
+      # Calls the the Convection::Model::Stack#get_status method
+      # if there is a rate limiting exception it sleeps and trys again.
+      def status_caller(count = 0.25)
+        get_status(cloud_name)
+      rescue Aws::CloudFormation::Errors::Throttling
+        sleep count
+        count += 0.25
+        status_caller(count)
+      end
+
+      # Calls The below methods in parallel
+      #   Convection::Model::Stack#get_resources
+      #   Convection::Model::Stack#get_template
+      #   Convection::Model::Stack#resource_attributes
+      #   Convection::Model::Stack#get_events
+      def resolver
+        hash = { :get_resources => nil, :get_template => nil, :resource_attributes => nil, :get_events => 1 }
+        threads = []
+        hash.each do |call_me, value|
+          threads << Thread.new do
```

Comment:
I think `get_resources` and `resource_attributes` need to be run series and can't be run in parallel because `get_resources` populates the `@resources` instance variable that `resource_attributes` iterates through.  I'd suggest auditing the code to see if there are other similar dependencies, as well as if any semaphores are needed.